graylog-storage-opensearch3/src/main/java/org/graylog/storage/opensearch3/SecurityAdapterOS.java

```diff
@@ -16,97 +16,59 @@
  */
 package org.graylog.storage.opensearch3;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.annotation.Nonnull;
 import jakarta.inject.Inject;
-import org.graylog.shaded.opensearch2.org.apache.http.HttpEntity;
-import org.graylog.shaded.opensearch2.org.apache.http.entity.ContentType;
-import org.graylog.shaded.opensearch2.org.apache.http.entity.StringEntity;
-import org.graylog.shaded.opensearch2.org.opensearch.client.Request;
 import org.graylog2.indexer.security.SecurityAdapter;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.opensearch.client.json.JsonData;
+import org.opensearch.client.opensearch.security.GetRoleMappingResponse;
+import org.opensearch.client.opensearch.security.PatchRoleMappingResponse;
+import org.opensearch.client.opensearch.security.RoleMapping;
 
-import java.util.ArrayList;
+import java.io.IOException;
 import java.util.List;
 
 /**
  * see: https://opensearch.org/docs/latest/security/access-control/api/
  */
 public class SecurityAdapterOS implements SecurityAdapter {
-    private static final Logger LOG = LoggerFactory.getLogger(SecurityAdapterOS.class);
 
-    final private PlainJsonApi jsonApi;
-    private final ObjectMapper objectMapper;
+    private final OfficialOpensearchClient client;
 
     @Inject
-    public SecurityAdapterOS(final ObjectMapper objectMapper,
-                             final PlainJsonApi jsonApi) {
-        this.objectMapper = objectMapper;
-        this.jsonApi = jsonApi;
+    public SecurityAdapterOS(OfficialOpensearchClient openSearchClient) {
+        this.client = openSearchClient;
     }
 
     @Override
     public Mapping getMappingForRole(final String role) {
-        final JsonNode result = jsonApi.perform(request("GET", "rolesmapping/" + role), "Unable to retrieve role mapping for role " + role);
-
-        final JsonNode fields = result.path(role);
         try {
-            return objectMapper.treeToValue(fields, Mapping.class);
-        } catch (JsonProcessingException e) {
+            final GetRoleMappingResponse response = client.sync().security().getRoleMapping(r -> r.role(role));
```

Comment:
If you wrap this with `client.execute`, you could use the old exception message and could get rid of the try-catch

Reply:
I went one step ahead and added a sync method doing the same thing, providing the client as an argument:

```
officialClient.sync(syncClient -> syncClient.security().createAllowlist(...))
```

Maybe we should hide the actual client and force all usages to go through the exception mapping?

Reply:
good idea, I was reluctant at first, because you might want to actually handle the exception yourself. but you can still do that in the lambda